dio/lib/src/dio_error.dart

```diff
@@ -2,35 +2,109 @@ import 'options.dart';
 import 'response.dart';
 
 enum DioErrorType {
-  /// It occurs when url is opened timeout.
-  connectTimeout,
+  /// Caused by a connection timeout.
+  connectionTimeout,
 
   /// It occurs when url is sent timeout.
   sendTimeout,
 
   ///It occurs when receiving timeout.
   receiveTimeout,
 
-  /// When the server response, but with a incorrect status, such as 404, 503...
-  response,
+  /// The [DioError] was caused by an incorrect status code as configured by
+  /// [ValidateStatus].
+  badResponse,
 
   /// When the request is cancelled, dio will throw a error with this type.
-  cancel,
+  requestCancelled,
```

Comment:
Users usually ignore canceled requests since they are meant to do it, which means they'll rely on the en. Maybe we can keep the naming at least for the least breaking.